### PR TITLE
New example.

### PR DESCRIPTION
**Pull request by PurgePJ**
Had to make this file due to the amount of people asking on how to make a help command to display all the available commands in their bot.

Simple solution and easy to understand.

https://i.imgur.com/rtfmHrj.png

### CODE REVIEW

**Comment by Tigerism** [comment]
This example should be consistent with the other examples

File: examples/helpCommandExample.lua [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		end
+	}
+}
+
+client:on('ready', function()
+	p(string.format('Logged in as %s', client.user.username))
+end)
+
+client:on("messageCreate", function(message)
+	local args = content:split(" ") -- split all arguments into a table
```
Comment (SinisterRectus) [inline examples/helpCommandExample.lua]: `content` is not defined.

File: examples/helpCommandExample.lua [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+	local command = commands[args[1]]
+	if command then -- ping or hello
+		command.exec(message) -- execute the command
+	end
+
+	if args[1] == "help" then -- display all the commands
```
Comment (SinisterRectus) [inline examples/helpCommandExample.lua]: Prefix?

File: examples/helpCommandExample.lua [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		local output = ""
+		for word, tbl in pairs(commands) do
+			output = output .. "Command: " .. word .. "\nDescription: " .. tbl.description .. "\n\n"
```
Comment (SinisterRectus) [inline examples/helpCommandExample.lua]: Multiple concatenations and concatenations in a loop are bad. Use a buffer table and concat with newlines after the loop.

File: examples/helpCommandExample.lua [hunk 4]
```diff
@@ -31,12 +31,12 @@ client:on("messageCreate", function(message)
 	end
 
 	if args[1] == prefix.."help" then -- display all the commands
-		local output = ""
+		local output = {""}
```
Comment (SinisterRectus) [inline examples/helpCommandExample.lua]: Empty string not necessary.

File: examples/helpCommandExample.lua [hunk 5]
```diff
@@ -31,12 +31,12 @@ client:on("messageCreate", function(message)
...
 		for word, tbl in pairs(commands) do
-			output = output .. "Command: " .. word .. "\nDescription: " .. tbl.description .. "\n\n"
+			table.insert(output, "Command: " .. word .. "\nDescription: " .. tbl.description .. "\n\n")
```
Comment (SinisterRectus) [inline examples/helpCommandExample.lua]: Terminating newline `\n` not necessary if you concat with newlines.

File: examples/helpCommandExample.lua [hunk 6]
```diff
@@ -31,12 +31,12 @@ client:on("messageCreate", function(message)
...
 		end
 
-		message:reply(output)
+		message:reply(table.concat(output))
```
Comment (SinisterRectus) [inline examples/helpCommandExample.lua]: Concat with `\n`.

**Comment by PurgePJ** [comment]
good

**Comment by SinisterRectus** [comment]
Thank you.

**Pull request merged**